charts/newrelic-infrastructure-v3/templates/_helpers.tpl

```diff
@@ -241,3 +241,13 @@ interval: 15s
 {{- end -}}
 {{- end -}}
 
+{{/*
+Renders a value that contains template.
+*/}}
+{{- define "newrelic.tplvalues.render" -}}
```

Comment:
still deciding about this

Reply:
Use the helpers in the init containers to have access to options like license, staging and such.

I think it could be useful.

Reply:
I don't see the point of limiting the initContainer `tpl` to `.context`.

Why wouldn't I want to have access to all the `.Values` and templates?

Reply:
Everything is passed, see here: https://github.com/newrelic/helm-charts/pull/714/files#diff-b2298e732a892ff75958aab46b0835f139eb0ed344eede9b00569040056a4d18R47

Reply:
I took it from here https://github.com/bitnami/charts/blob/master/bitnami/common/templates/_tplvalues.tpl while looking around possible ways to do so.

I am not sure it worth the additional complexity. Should we just have `{{- tpl (.value | toYaml) .context }}`. I am not sure we have any use-case for witch we need to support strings (apart from custom attributes)

Reply:
If you are giving the ability to render things this makes things difficult. For example, if you want to `template "newrelic.fullname" .` you have to `template "newrelic.fullname" .context` or `template "newrelic.fullname" $` to properly render.

That is the thing that worries me the most: Having a `value` and `context` instead of having directly `.Values` and that's it.

I would not have in mind that the initContainers could be strings. That is a really rare use case that I would not like to support. On the other hand, I could find edge cases where I would want to have access to the values and the templating in the init containers so templating the YAML in `.Values.initContainers` sounds cool.